src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkWatchListService.java

```diff
@@ -73,27 +84,74 @@ public void removeWatch(Endpoint endpoint) {
         endpoints.remove(endpoint);
     }
 
+    /**
+     * Handle update slow discovery poll flag request.
+     */
+    public void updateSlowPoll(Endpoint endpoint, boolean isEnabled) {
+        log.info("Watch-list service receive update slow poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
+                endpointContext.toBuilder().slowPollEnabled(isEnabled).build());
+    }
+
+    @VisibleForTesting
+    void updateUnhurriedPoll(Endpoint endpoint, boolean isEnabled, long currentTime) {
+        log.info("Watch-list service receive update unhurried poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
+                endpointContext.toBuilder().unhurriedPollEnabled(isEnabled).build());
+
+        if (!isEnabled) { // when another mechanism used to determine ISL status is disabled
```

Comment:
Why we are not doing same actions on disabling "slow" mode?

Reply:
This is not necessary for our case (for ports not belonging to ISL). If we need to use this mode for other cases, it is not difficult to fix it.